// @sap/di.code-validation.js/tasks/code-validation-task.js
'use strict';
var path = require('path');
module.exports = function(grunt) {
    grunt.registerTask('di.code-validation.js', function () {
        grunt.loadNpmTasks('grunt-mkdir');
        grunt.loadNpmTasks('di.code-validation.grunt');

        var pkg = grunt.file.readJSON('package.json');
        var root = path.join(pkg.name);
        var tmpDir = path.join('tmp-lint');

        grunt.config("mkdir", {
            tmpDir: {
                options: {
                    create: [tmpDir]
                }
            }
        });

        grunt.config("codevalidation", {
            "di.code-validation.js": {
                options: {
                    projectPath: process.cwd(),
                    pathPrefix: path.join(path.sep, root),
                    ignoredPaths: ['node_modules', tmpDir],
                    reporter: "problems_reporter",
                    reporterOptions: {outputFile: path.join(tmpDir, "di.code-validation.js_issues.json")},
                    validators: {
                        "di.code-validation.js": {
                            extensions: [".js"]
                        }
                    }
                }
            }
        });

        grunt.task.run(['mkdir:tmpDir', 'codevalidation:di.code-validation.js']);

    });
};
